internal/services/notification/notification_test.go

```diff
@@ -174,6 +178,149 @@ func TestRunWebhookDelivery(t *testing.T) {
+
+	err := ns.runEventsHandler(ctx)
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+
+	lastRunEventSequenceValue = getLastRunEventSequenceValue(t, ctx, ns)
+	if lastRunEventSequenceValue != expectedLastRunEventSequenceValue {
+		t.Fatalf("expected %d last run event got: %d", expectedLastRunEventSequenceValue, lastRunEventSequenceValue)
+	}
+
+	// test new events
+	cleanRunWebhooks(t, ctx, ns)
+
+	runEvents := make([]*rstypes.RunEvent, 2)
+	runEvents[0] = generateRunEvent(3, rstypes.RunPhaseChanged)
+	runEvents[1] = generateRunEvent(4, rstypes.RunPhaseChanged)
+
+	runEventsSender.runEvents.addRunEvent(runEvents[0])
+	runEventsSender.runEvents.addRunEvent(runEvents[1])
+
+	err = ns.runEventsHandler(ctx)
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+
+	runWebhooks := getRunWebhooks(t, ctx, ns)
+	if len(runWebhooks) != len(runEvents) {
+		t.Fatalf("expected %d runWebhooks got: %d", len(runEvents), len(runWebhooks))
+	}
+
+	for i := range runEvents {
+		runWebhook := ns.generatewebhook(ctx, runEvents[i])
+		webhookPayload, err := json.Marshal(runWebhook)
+		if err != nil {
+			t.Fatalf("unexpected err: %v", err)
+		}
+
+		if !bytes.Equal(runWebhooks[i].Payload, webhookPayload) {
+			t.Fatalf("expected %s run webhook payload got: %s", runWebhooks[i].Payload, webhookPayload)
+		}
+	}
+
+	expectedLastRunEventSequenceValue = 4
+	lastRunEventSequenceValue = getLastRunEventSequenceValue(t, ctx, ns)
+	if lastRunEventSequenceValue != expectedLastRunEventSequenceValue {
+		t.Fatalf("expected %d last run event sequence got: %d", expectedLastRunEventSequenceValue, lastRunEventSequenceValue)
+	}
+}
+
+func cleanRunWebhooks(t *testing.T, ctx context.Context, ns *NotificationService) {
+	err := ns.d.Do(ctx, func(tx *sql.Tx) error {
+		var err error
+		runWebhookDeliverys, err := ns.d.GetRunWebhookDeliveriesAfterSequence(tx, 0, "", 0)
+		if err != nil {
+			return errors.WithStack(err)
+		}
+
+		for _, runWebhookDelivery := range runWebhookDeliverys {
+			if err := ns.d.DeleteRunWebhookDelivery(tx, runWebhookDelivery.ID); err != nil {
+				return errors.WithStack(err)
+			}
+		}
+
+		runWebhooks, err := ns.d.GetRunWebhooks(tx, 0)
+		if err != nil {
+			return errors.WithStack(err)
+		}
+
+		for _, runWebhook := range runWebhooks {
+			if err := ns.d.DeleteRunWebhook(tx, runWebhook.ID); err != nil {
+				return errors.WithStack(err)
+			}
+		}
+
+		return nil
+	})
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+}
+
+func generateRunEvent(sequence uint64, runEventType rstypes.RunEventType) *rstypes.RunEvent {
+	return &rstypes.RunEvent{
+		Sequence:     sequence,
+		RunEventType: runEventType,
+		DataVersion:  rstypes.RunEventDataVersion,
+		Data:         &rstypes.RunEventData{},
+	}
+}
+
+func getLastRunEventSequenceValue(t *testing.T, ctx context.Context, ns *NotificationService) uint64 {
+	var lastRunEventSequence uint64
+
+	err := ns.d.Do(ctx, func(tx *sql.Tx) error {
+		var err error
```

Comment:
this isn't needed if you're populating lastRunEventSequence inside the transaction.